# Handle empty string results when calling createLikelySubtagsString() for _uloc_minimizeSubtags().
#
# ICU bug: https://ssl.icu-project.org/trac/ticket/13786

diff --git a/intl/icu/source/common/loclikely.cpp b/intl/icu/source/common/loclikely.cpp
--- a/intl/icu/source/common/loclikely.cpp
+++ b/intl/icu/source/common/loclikely.cpp
@@ -1051,17 +1051,17 @@ _uloc_minimizeSubtags(const char* localeID,
                 0,
                 sink,
                 err);
         }
 
         if(U_FAILURE(*err)) {
             goto error;
         }
-        else if (uprv_strnicmp(
+        else if (!tagBuffer.isEmpty() && uprv_strnicmp(
                     maximizedTagBuffer.data(),
                     tagBuffer.data(),
                     tagBuffer.length()) == 0) {
 
             createTagString(
                         lang,
                         langLength,
                         NULL,
@@ -1096,17 +1096,17 @@ _uloc_minimizeSubtags(const char* localeID,
                 0,
                 sink,
                 err);
         }
 
         if(U_FAILURE(*err)) {
             goto error;
         }
-        else if (uprv_strnicmp(
+        else if (!tagBuffer.isEmpty() && uprv_strnicmp(
                     maximizedTagBuffer.data(),
                     tagBuffer.data(),
                     tagBuffer.length()) == 0) {
 
             createTagString(
                         lang,
                         langLength,
                         script,
